packages/@aws-cdk/aws-glue/lib/connection.ts

```diff
@@ -0,0 +1,193 @@
+import * as ec2 from '@aws-cdk/aws-ec2';
+import * as cdk from '@aws-cdk/core';
+import * as constructs from 'constructs';
+import { CfnConnection } from './glue.generated';
+
+/**
+ * The type of the glue connection
+ *
+ * If you need to use a connection type that doesn't exist as a static member, you
+ * can instantiate a `ConnectionType` object, e.g: `new ConnectionType('NEW_TYPE')`.
+ */
+export class ConnectionType {
+  /** A list of all known `ConnectionType`s. */
+  public static readonly ALL = new Array<ConnectionType>();
+
+  /**
+   * Designates a connection to a database through Java Database Connectivity (JDBC).
+   */
+  public static readonly JDBC = new ConnectionType('JDBC');
+
+  /**
+   * Designates a connection to an Apache Kafka streaming platform.
+   */
+  public static readonly KAFKA = new ConnectionType('KAFKA');
+
+  /**
+   * Designates a connection to a MongoDB document database.
+   */
+  public static readonly MONGODB = new ConnectionType('MONGODB');
+
+  /**
+   * Designates a network connection to a data source within an Amazon Virtual Private Cloud environment (Amazon VPC).
+   */
+  public static readonly NETWORK = new ConnectionType('NETWORK');
+
+  /**
+   * The name of this ConnectionType, as expected by Connection resource.
+   */
+  public readonly name: string;
+
+  constructor(name: string) {
+    this.name = name;
+    ConnectionType.ALL.push(this);
+  }
+
+  /**
+   * The connection type name as expected by Connection resource.
+   */
+  public toString(): string {
+    return this.name;
+  }
+}
+
+/**
+ * Interface representing a created or an imported {@link Connection}
+ */
+export interface IConnection extends cdk.IResource {
+  /**
+   * The name of the connection
+   * @attribute
+   */
+  readonly connectionName: string;
+
+  /**
+   * The ARN of the connection
+   * @attribute
+   */
+  readonly connectionArn: string;
+}
+
+/**
+ * Attributes for importing {@link Connection}
+ */
+export interface ConnectionAttributes {
+  /**
+   * The name of the connection
+   */
+  readonly connectionName: string;
+}
+
+/**
+ * Construction properties for {@link Connection}
+ */
+export interface ConnectionProps {
+  /**
+   * The name of the connection
+   * @default cloudformation generated name
+   */
+  readonly connectionName?: string;
+
+  /**
+   * The description of the connection.
+   * @default no description
+   */
+  readonly description?: string;
+
+  /**
+   * The type of the connection
+   */
+  readonly connectionType: ConnectionType;
+
+  /**
+   *  Key-Value pairs that define parameters for the connection.
+   *  @default empty properties
+   */
+  readonly connectionProperties?: { [key: string]: string };
+
+  /**
+   * A list of criteria that can be used in selecting this connection.
+   * @default no match criteria
+   */
+  readonly matchCriteria?: string[];
```

Comment:
An example and/or documentation of these would be good.

Reply:
Added a further comment on where this can be used